src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -454,18 +460,130 @@ export class HeaderFilter<T extends Slick.SlickData> {
+                    let gridColumnMapArray = store.get(
+                        this.queryResultState.state.uri!,
+                    ) as GridColumnMap[];
+                    gridColumnMapArray = this.clearFilterValues(
+                        gridColumnMapArray,
+                        columnDef.id!,
+                    );
+                    store.set(
+                        this.queryResultState.state.uri,
+                        gridColumnMapArray,
+                    );
+                }
+            }
         } else {
             columnFilterState = {
                 columnDef: this.columnDef.id!,
                 filterValues: this.columnDef.filterValues!,
                 sorted: this.columnDef.sorted,
             };
         }
-        this.updateState(columnFilterState);
+
+        this.updateState(columnFilterState, columnDef.id!);
+    }
+
+    private updateState(newState: ColumnFilterState, columnId: string): void {
+        let newStateArray: GridColumnMap[];
+        //Check if there is any existing filter state
+        if (!this.queryResultState.state.uri) {
+            console.log("no uri set for query result state");
+            return;
+        }
+        const currentFiltersArray = store.get(
+            this.queryResultState.state.uri,
+        ) as GridColumnMap[];
+        if (currentFiltersArray) {
+            newStateArray = this.combineFilters(
+                currentFiltersArray,
+                newState,
+                columnId,
+            );
+        } else {
+            newStateArray = [{ [this.gridId]: [{ [columnId]: [newState] }] }];
+        }
+        store.set(this.queryResultState.state.uri, newStateArray);
+    }
+
+    private clearFilterValues(
+        gridFiltersArray: GridColumnMap[],
+        columnId: string,
+    ) {
+        const targetGridFilters = gridFiltersArray.find(
+            (gridFilters) => gridFilters[this.gridId],
+        );
+
+        // Return original array if gridId is not found
+        if (!targetGridFilters) {
+            return gridFiltersArray;
+        }
+
+        // Iterate through each ColumnFilterMap and clear filterValues for the specified columnId
+        for (const columnFilterMap of targetGridFilters[this.gridId]) {
+            if (columnFilterMap[columnId]) {
+                columnFilterMap[columnId] = columnFilterMap[columnId].map(
+                    (filterState) => ({
+                        ...filterState,
+                        filterValues: [],
+                    }),
+                );
+            }
+        }
+
+        return [...gridFiltersArray];
     }
 
-    private updateState(newState: ColumnFilterState) {
-        this.queryResultState.provider.setFilterState(newState);
+    /**
+     * Combines two GridColumnMaps into one, keeping filters separate for different gridIds and removing any duplicate filterValues within the same column id
+     * @param currentFiltersArray
```

Comment:
is `currentFiltersArray` for all filters across all grids, or just across one grid?  (please add that info to the `@param currentFiltersArray` docstring)

Reply:
it's across all grids

Reply:
it's across all grids

Reply:
`@param currentFiltersArray <doc string please!>`